docs/design.md

```diff
@@ -138,7 +138,17 @@ upgrades, but must be configured at build time to choose one of these two
 strategies.
 
 In addition to the two image slots, the boot loader requires a scratch area to
-allow for reliable image swapping.
+allow for reliable image swapping. The scratch area must have a size that is
+enough to store at least the largest sector that is going to be swapped. Many
+devices have small linear sized flash sectors, eg 4K, while others have variable
+sized sectors where the largest sectors might be 128K or 256K, so the scratch
+must be big enough to store that. The scratch is only ever used when swapping
+firmware, which means only when doing an upgrade. Given that, there are two main
+reasons for using a larger size for the scratch. One is that the speed of an
+upgrade is proportional to the size, because larger size allows for less swap
+operations. Second is that flash wear will be more evenly distributed, because
```

Comment:
Overall I like this change. I think that it would be an improvement to quantify the wear on the flash. People's biggest question is always "is this big enough", and a quantitative description in terms of how much it will reduce the lifetime of the flash will help them make their own decisions.

Here is an example where I described the tradeoff for nRF5:

https://github.com/zephyrproject-rtos/zephyr/pull/1307#discussion_r136318044

Do you think you can include something like that, with concrete numbers or a formula?

Reply:
Done!